project/src/ui/master_control/ProfileConfig.gd

```diff
@@ -19,6 +19,7 @@ class_name ProfileConfig
 
 var profile_name: String = "No Name"
 var environment: String = "playground/Playground"
+var compiler: CompilerInformation = CompilerInformation.new()
 var slots: Array = []
 
 func type_info() -> Dictionary:
```

Comment:
I believe you need to add some type info to this for serialization to not fail,
`"compiler": CompilerInformation`
You'll have to test it though

Reply:
When I saved the profile, it was not a problem at least. I thought the serialization is working due to CompilerInformation inheriting Reference. And I saw a serialization method for that. Or do you mean something else?

Reply:
References get turned into a Dictionary, but if you want to de-serialize it back it needs a real type.
Easiest way to test it is by saving a profile with a non default compiler and then restarting the app.

Reply:
I tried it out and it worked well.